starknet-signers/src/local_wallet.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{Infallible, Signer, SigningKey, VerifyingKey};
+
+use async_trait::async_trait;
+use starknet_core::{
+    crypto::{EcdsaSignError, Signature},
+    types::UnsignedFieldElement,
+};
+
+pub struct LocalWallet {
+    private_key: SigningKey,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum SignError {
+    #[error(transparent)]
+    EcdsaSignError(EcdsaSignError),
+}
+
+impl LocalWallet {
+    pub fn from_signing_key(key: SigningKey) -> Self {
+        key.into()
+    }
+}
+
+#[async_trait]
+impl Signer for LocalWallet {
+    type GetPublicKeyError = Infallible;
```

Comment:
`Infallible` exists just to comply with the API and as to mark that this one can never fail?

Reply:
Yep just to comply with the `Signer` API, as that's a generic one and other implementations will be fallible due to a lot of reasons. This is similar to `std::convert::Infallible`.